test/WebJobs.Script.Tests/Workers/Rpc/JobHostRpcWorkerChannelManagerTests.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.WebJobs.Script.Workers.Rpc;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.Workers.Rpc
+{
+    public class JobHostRpcWorkerChannelManagerTests
+    {
+        private static readonly ILoggerFactory _loggerFactory = MockNullLoggerFactory.CreateLoggerFactory();
+        private Mock<IRpcWorkerChannel> _workerChannelJava1 = new Mock<IRpcWorkerChannel>();
+        private Mock<IRpcWorkerChannel> _workerChannelJava2 = new Mock<IRpcWorkerChannel>();
+        private Mock<IRpcWorkerChannel> _workerChannelJs1 = new Mock<IRpcWorkerChannel>();
+        private Mock<IRpcWorkerChannel> _workerChannelJs2 = new Mock<IRpcWorkerChannel>();
+        private Mock<IRpcWorkerChannel> _workerChannelPs1 = new Mock<IRpcWorkerChannel>();
+        private Mock<IRpcWorkerChannel> _workerChannelPs2 = new Mock<IRpcWorkerChannel>();
+        private JobHostRpcWorkerChannelManager _jobHostRpcWorkerChannelManager = new JobHostRpcWorkerChannelManager(_loggerFactory);
+
+        public JobHostRpcWorkerChannelManagerTests()
+        {
+            _workerChannelJava1.Setup(wc => wc.Id).Returns("java1");
+            _workerChannelJava2.Setup(wc => wc.Id).Returns("java2");
+
+            _workerChannelJs1.Setup(wc => wc.Id).Returns("js1");
+            _workerChannelJs2.Setup(wc => wc.Id).Returns("js2");
+
+            _workerChannelPs1.Setup(wc => wc.Id).Returns("ps1");
+            _workerChannelPs2.Setup(wc => wc.Id).Returns("ps2");
+        }
+
+        [Fact]
+        public void TestAddChannel()
+        {
+            _jobHostRpcWorkerChannelManager.ShutdownChannels();
+            _jobHostRpcWorkerChannelManager.AddChannel(_workerChannelJava1.Object, "java");
+            _jobHostRpcWorkerChannelManager.AddChannel(_workerChannelPs1.Object, "powershell");
+            _jobHostRpcWorkerChannelManager.AddChannel(_workerChannelJs1.Object, "node");
+
+            Assert.Equal(_jobHostRpcWorkerChannelManager.GetChannels().Count(), 3);
```

Comment:
use named parameters due to the ambiguity in same type arguments order.

Reply:
Not followed anywhere in function code, will wait for owners response on this